crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -284,49 +289,77 @@ impl SolverTimeout {
     }
 }
 
-/// The solution score. This is often referred to as the "objective value".
-#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
-pub struct Score(pub eth::U256);
+/// Carries information how the score should be calculated.
+#[derive(Debug, Clone)]
+pub enum Score {
+    Solver(eth::U256),
+    Discount(eth::U256),
+    RiskAdjusted {
+        success_probability: SuccessProbability,
+        gas_amount: Option<eth::U256>,
+    },
+}
 
-impl From<Score> for eth::U256 {
-    fn from(value: Score) -> Self {
-        value.0
-    }
+/// Represents the probability that a solution will be successfully settled.
+#[derive(Debug, Clone)]
+pub enum SuccessProbability {
+    /// Probability known and is equal to the given value.
+    Value(f64),
+    /// Probability unknown and should be computed by the protocol using the
+    /// given parameters.
+    Params {
+        gas_amount_factor: f64,
+        gas_price_factor: f64,
+        nmb_orders_factor: f64,
+        intercept: f64,
+    },
 }
 
-impl From<eth::U256> for Score {
-    fn from(value: eth::U256) -> Self {
-        Self(value)
+/// Represents score together with the way it was calculated.
+#[derive(Copy, Clone, Debug)]
+pub enum CalculatedScore {
```

Comment:
It's needed for logging in the driver [here](https://github.com/cowprotocol/services/blob/optimal_bidding-colocation/crates/driver/src/domain/competition/mod.rs#L164-L167).
I guess it is nice to keep it for now.

Reply:
Personally, I think carrying this information around in the domain just for logging is a smell. Can't we just log this information when/where the score is computed instead of leaking this?

Reply:
Removed.